File: src/api/schemas/retrieve.py
"""检索接口 Pydantic 模型"""

from typing import Literal, Optional

from pydantic import BaseModel, Field


class RetrieveRequest(BaseModel):
    question: str
    top_k: int = Field(default=10, ge=1, le=50)
    search_mode: Literal["vector", "bm25", "hybrid"] = "vector"
    show_prompt: bool = False
    dataset_ids: Optional[list[str]] = None
    doc_ids: Optional[list[str]] = None
    doc_names: Optional[list[str]] = None


class ChunkMetadataResult(BaseModel):
    chunk_id: str
    chunk_type: str
    filename: str
    page: int
    pages: list[int] = []
    chunk_index: int
    char_count: int
    created_at: str
    doc_id: str


class ChunkScores(BaseModel):
    vector_score: float = 0.0
    bm25_score: float = 0.0
    rrf_score: Optional[float] = None


class RetrievedChunkResult(BaseModel):
    rank: int
    metadata: ChunkMetadataResult
    full_text: str
    scores: ChunkScores
    image_urls: list[str] = []


class RetrieveResponse(BaseModel):
    question: str
    search_mode: str
    total_retrieved: int
    retrieval_ms: int
    chunks: list[RetrievedChunkResult]
    prompt: Optional[str] = None

File: web/src/api/retrieve.ts
import request from './request'

export interface RetrieveRequest {
  question: string
  top_k?: number
  search_mode?: 'vector' | 'bm25' | 'hybrid'
  dataset_ids?: string[]
  doc_ids?: string[]
  doc_names?: string[]
}

export interface ChunkScores {
  vector_score: number
  bm25_score: number
  rrf_score: number | null
}

export interface ChunkMetadataResult {
  chunk_id: string
  chunk_type: string
  filename: string
  page: number
  pages: number[]
  chunk_index: number
  char_count: number
  created_at: string
  doc_id: string
}

export interface RetrievedChunkResult {
  rank: number
  metadata: ChunkMetadataResult
  full_text: string
  scores: ChunkScores
  image_urls: string[]
}

export interface RetrieveResponse {
  question: string
  search_mode: string
  total_retrieved: number
  retrieval_ms: number
  chunks: RetrievedChunkResult[]
}

export const retrieveChunks = (data: RetrieveRequest) =>
  request.post<any, RetrieveResponse>('/retrieve', data)
